feat: add match editing

bjarkehs opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Moderators can edit match details (teams, scores, season) and delete matches from the matches list via inline Edit/Delete dialogs; actions trigger MMR recalculation and show success or warning messages.
  * Matches list now surfaces current season context for edits.

* **Bug Fixes / Validation**
  * Form validation enforces four unique players and non-empty scores; improved error handling and user-friendly messages.

* **Style**
  * Adjusted team member alignment in match display.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

bjarkehs committed:
feat: add match editing

bjarkehs commented on api/MMRProject.Api/Services/MatchesService.cs:
```diff
@@ -519,4 +529,91 @@ private async Task ClearMMRCalculations(long seasonId, long? fromMatchId)
+        }
+
+        var uniquePlayers = new HashSet<long>
+        {
+            request.Team1.Member1,
+            request.Team1.Member2,
+            request.Team2.Member1,
+            request.Team2.Member2
+        };
+
+        if (uniquePlayers.Count != 4)
+        {
+            throw new InvalidArgumentException("Players must be unique");
+        }
+
+        var players = await dbContext.Players.Where(x => uniquePlayers.Contains(x.Id)).ToListAsync();
+
+        if (players.Count != uniquePlayers.Count)
+        {
+            throw new InvalidArgumentException("Not all players were found");
+        }
+
+        match.TeamOne!.PlayerOneId = request.Team1.Member1;
+        match.TeamOne.PlayerTwoId = request.Team1.Member2;
+        match.TeamOne.Score = request.Team1.Score;
+        match.TeamOne.Winner = request.Team1.Score > request.Team2.Score;
+        match.TeamOne.UpdatedAt = DateTime.UtcNow;
+
+        match.TeamTwo!.PlayerOneId = request.Team2.Member1;
+        match.TeamTwo.PlayerTwoId = request.Team2.Member2;
+        match.TeamTwo.Score = request.Team2.Score;
+        match.TeamTwo.Winner = request.Team2.Score > request.Team1.Score;
+        match.TeamTwo.UpdatedAt = DateTime.UtcNow;
+
+        match.UpdatedAt = DateTime.UtcNow;
+
+        await dbContext.SaveChangesAsync();
+
+        return match;
+    }
```
The admin editing the matches will do it for now

bjarkehs commented on api/MMRProject.Api/Services/MatchesService.cs:
```diff
@@ -519,4 +529,91 @@ private async Task ClearMMRCalculations(long seasonId, long? fromMatchId)
+        return match;
+    }
+
+    public async Task DeleteMatch(long matchId)
+    {
+        var match = await dbContext.Matches
+            .Include(x => x.TeamOne)
+            .Include(x => x.TeamTwo)
+            .FirstOrDefaultAsync(x => x.Id == matchId);
+
+        if (match == null)
+        {
+            throw new InvalidArgumentException("Match not found");
+        }
+
+        var latestSeason = await dbContext.Seasons.OrderByDescending(x => x.Id).FirstOrDefaultAsync();
+        if (latestSeason?.Id != match.SeasonId)
+        {
+            throw new InvalidArgumentException("Only matches in the current season can be deleted");
+        }
+
+        match.DeletedAt = DateTime.UtcNow;
+        match.TeamOne!.DeletedAt = DateTime.UtcNow;
+        match.TeamTwo!.DeletedAt = DateTime.UtcNow;
+
+        await dbContext.SaveChangesAsync();
+    }
```
We are having the frontend ensure this happens for now

bjarkehs committed:
implement review comments

madsthom commented on api/MMRProject.Api/Controllers/Admin/AdminMatchController.cs:
```diff
@@ -0,0 +1,35 @@
+) : ControllerBase
+{
+    [HttpPut("{matchId}")]
+    public async Task<ActionResult<MatchDetailsV2>> UpdateMatch([FromRoute] long matchId, [FromBody] UpdateMatchRequest request)
```
Du vil ikke lave en recalculation når der updateres/slettes en kamp. Det skal man huske selv at gøre, right?

bjarkehs replied:
Jep

madsthom commented on frontend/src/routes/admin/matches/+page.svelte:
```diff
@@ -146,3 +248,176 @@
+				<Button type="button" variant="outline" onclick={() => showEditDialog = false} disabled={isEditingMatch}>Cancel</Button>
+				<Button type="submit" disabled={!isEditFormValid || isEditingMatch}>
+					{isEditingMatch ? 'Updating...' : 'Save Changes'}
+				</Button>
+			</Dialog.Footer>
+		</form>
+	</Dialog.Content>
+</Dialog.Root>
+
+<Dialog.Root open={showDeleteDialog} onOpenChange={(open) => showDeleteDialog = open}>
+	<Dialog.Content>
+		<Dialog.Header>
+			<Dialog.Title class="flex items-center gap-2">
+				<AlertTriangle class="h-5 w-5 text-destructive" />
+				Delete Match
+			</Dialog.Title>
+			<Dialog.Description>
+				Are you sure you want to delete Match #{editingMatch?.matchId}? This action will automatically recalculate all subsequent matches.
+			</Dialog.Description>
+		</Dialog.Header>
+		<form bind:this={deleteFormElement} method="POST" action="?/deleteMatch" use:enhance={() => {
+			return async ({ result, update }) => {
+				await update();
+				isDeletingMatch = false;
+				if (result.type === 'success') {
+					window.location.reload();
+				}
+			};
+		}}>
+			<input type="hidden" name="matchId" value={editingMatch?.matchId ?? ''} />
+			<Dialog.Footer class="gap-2">
+				<Button type="button" variant="outline" onclick={() => showDeleteDialog = false} disabled={isDeletingMatch}>Cancel</Button>
+				<Button type="button" variant="destructive" onclick={handleDeleteConfirm} disabled={isDeletingMatch}>
+					<Trash2 class="h-4 w-4 mr-2" />
+					{isDeletingMatch ? 'Deleting...' : 'Delete Match'}
+				</Button>
+			</Dialog.Footer>
+		</form>
+	</Dialog.Content>
+</Dialog.Root>
```
Donno om dette er vigtigt 🫨

bjarkehs committed:
fix: keep delete dialog open on errors and show error messages

Centralize deletion state management in use:enhance callback to ensure dialog remains open when server errors occur, allowing users to see error messages and retry or cancel.
